Tests/IGListBatchUpdateDataTests.m

```diff
@@ -123,14 +123,28 @@ - (void)test_whenMovingSections_withMoveFromConflictWithDelete_thatResultDropsTh
     IGListBatchUpdateData *result = [[IGListBatchUpdateData alloc] initWithInsertSections:indexSet(@[])
                                                                            deleteSections:indexSet(@[@2])
                                                                              moveSections:[NSSet setWithArray:@[newMove(2, 6), newMove(0, 2)]]
-                                                                         insertIndexPaths:[NSArray new]
-                                                                         deleteIndexPaths:[NSArray new]
-                                                                           moveIndexPaths:[NSArray new]];
+                                                                         insertIndexPaths:@[]
+                                                                         deleteIndexPaths:@[]
+                                                                           moveIndexPaths:@[]];
     XCTAssertEqual(result.deleteSections.count, 1);
     XCTAssertEqual(result.moveSections.count, 1);
     XCTAssertEqual(result.insertSections.count, 0);
     XCTAssertEqualObjects(result.deleteSections, indexSet(@[@2]));
     XCTAssertEqualObjects(result.moveSections.anyObject, newMove(0, 2));
 }
 
+- (void)test_whenDuplicateDeleteIndexPaths_thatDeletesStaggered {
+    IGListBatchUpdateData *result = [[IGListBatchUpdateData alloc] initWithInsertSections:indexSet(@[])
+                                                                           deleteSections:indexSet(@[])
+                                                                             moveSections:[NSSet new]
+                                                                         insertIndexPaths:@[]
+                                                                         deleteIndexPaths:@[newPath(2, 0), newPath(2, 0), newPath(2, 1), newPath(2, 4)]
+                                                                           moveIndexPaths:@[]];
+    XCTAssertEqual(result.deleteIndexPaths.count, 4);
+
+    NSArray *expected = @[newPath(2, 0), newPath(2, 1), newPath(2, 3), newPath(2, 4)];
```

Comment:
What about (2,2)? Is it going to be deleted? Won't that be an internal inconsistency since it really wasn't deleted and there is no corresponding insert to ensure that the correct number of items is in the end?